File: ELEC881_Tabnine/src/No Description Prompt/Python/q8-rotate-function.py
class Solution(object):
    def maxRotateFunction(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        n = len(nums)
        total, prefix_sum = 0, sum(nums)
        for i in range(n):
            total += i * nums[i]

        max_sum = total
        for i in range(n-1, -1, -1):
            total = total - prefix_sum + n * nums[i]
            max_sum = max(max_sum, total)

        return max_sum - n * max(nums)
        

File: ELEC881_Tabnine/src/No Description Prompt/Java script (JS)/q8-rotate-function.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxRotateFunction = function(nums) {
    let sum = 0;
    let maxSum = 0;
    let prefixSum = 0;
    for (let i = 0; i < nums.length; i++) {
        sum += nums[i];
        prefixSum += i * nums[i];
    }
    maxSum = prefixSum;
    for (let i = 1; i < nums.length; i++) {
        prefixSum = prefixSum - sum + nums.length * nums[i - 1];
        maxSum = Math.max(maxSum, prefixSum);
    }
    return maxSum;
};